documentation/docs/configuration/parameters.md

```diff
@@ -0,0 +1,142 @@
+* Value
+* Environment
+* Reference
+
+## Value Parameter
+
+The value parameter is the simplest form of parameter. Besides the
+`type` property, it offers also `value`. You can define whatever you
+like as the value. Even nested maps. This value is then accessible in
+the template file.
+
+Since `values` are the most common type of parameter, there is also a
+special short form syntax to define them. If your parameter is neither
+an array, nor an map, you can simply provide the value.
+
+Example:
+
+```yaml
+parameters:
+  threshold: 15
+  complexThreshold:
+    type: value
+    value:
+        amount: 15
+        unit: sec
+```
+
+In the template of this config you could then access the `threshold`
+parameter via `{ .threshold }`. To access e.g. the `amount` of the
+`complexThreshold` you could use `{ .complexThreshold.amount }`.
+
+## Environment Parameter
+
+Parameters of type `environment` allow you to reference a environment
+variable. The name of the env variable to reference is defined via a
+`name` property. It is also possible to provide a default value, should
+the env variable not be present. This can be done via the `default`
+property.
+
+If the `default` property is not set and the env variable is missing,
+the parameter cannot be resolved. This will fail the deployment.
+**Note** this is only the case, if the paramter is relevant to being
+deployed. Parameters not referenced by the config to deploy are not
+evaluated.
+
+Example:
+
+```yaml
+parameters:
+  owner:
+    type: environment
+    name: OWNER
+    default: "-"
+  target:
+    type: environment
+    name: TARGET
+```
+
+In this example, the `owner` parameter will evaluate to whatever value the
+`OWNER` env variable is set to. If the env variable is not present, it
+will evaluate to value `-`.
+
+The `target` parameter will evaluate to the value of the `TARGET` env variable.
+It will fail the deployment, if the variable is not set at deploy time.
+
+## Reference Parameter
+
+Since it is often required to reference some form of property of another
+configuration, monaco offers a special reference parameter. This parameter
+allows one configuration to depend on pretty much any parameter of another
+config. To archive this, one has to specify the `project`, `api`,`config` and
+`property` properties, to tell monaco where it gets its value. There is
+also a short notation, which is an array. The syntax for this short
+notation is like `[ "{project name}", "{api}", "{config name}", "{property}"" ]`
+(note that the values in `{}` have to be replaced with real values, like
+`["project-1", "management-zone", "main", "id"]`). Property can be any
+parameter of the target config, the `name` or the `id`. `id` in this case
+is the one from the dynatrace api. Monaco will make sure, that the deployment
+of configuration is ordered and that the dependant config is deployed first.
+
+If you configure a cycle of dependencies, the deployment will fail.
```

Comment:
very off hand comment for such an important concept as "no cyclic dependencies!" - which is logical for most devs, but likely not all users. 
Consider giving that it's own section - maybe we should add an FAQ style "why is monaco yelling about cyclic dependencies?"